http/src/main/java/io/micronaut/http/HttpStatus.java

```diff
@@ -262,10 +280,37 @@ public static HttpStatus valueOf(int code) {
             case 522:
                 return CONNECTION_TIMED_OUT;
             default:
-                throw new IllegalArgumentException("Invalid HTTP status code: " + code);
+                return custom(code, genericReason(code));
         }
     }
 
+    /**
+     * The status for the given code and reason.
+     *
+     * @param statusCode The status code
+     * @param reason The reason. If null, a generic reason, same as {@link HttpStatus#valueOf(int)}, is provided
+     * @return The {@link HttpStatus}
+     */
+    public static HttpStatus custom(int statusCode, String reason) {
+        if (Objects.isNull(reason)) {
+            return valueOf(statusCode);
+        }
+        CUSTOM_STATUS.code = statusCode;
+        CUSTOM_STATUS.reason = reason;
+        return CUSTOM_STATUS;
+    }
```

Comment:
This approach will not work, it is not thread safe and sharing static state is not viable

Reply:
From this [StackOverflow post](https://stackoverflow.com/a/9614325/12591546) it seems that it is **not** possible to create a new instance of an enum, which I thought could solve the shared state issue. Anyway, I believe that an enum is still required for the annotation [`@Status`](https://github.com/micronaut-projects/micronaut-core/blob/2.5.x/http/src/main/java/io/micronaut/http/annotation/Status.java) to work, since the [Java Language Specification](https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.6.1) doesn't allow classes or interfaces as a member type.

I have made changes to the PR to meet two criteria:

- Keep the annotation `@Status` functional
- Make the implementation thread safe

I changed many files, the big-picture idea is shown in the following diagram:

<p align="center">
<img width="500" alt="Screen Shot 2021-06-28 at 0 31 08" src="https://user-images.githubusercontent.com/42811720/123585719-46129580-d7a9-11eb-99f0-4457a08e32a5.png">
</p>

**This approach has an inconvenience**: Since the name of the enum is changed, `@Status` is now used as follows:

```java
@Status(HttpStatusStandard.CREATED)
```

Instead of:

```java
@Status(HttpStatus.CREATED)
```

What are your thoughts on this approach?

Reply:
A possible workaround (without a huge refactor) is to ship the same HttpStatus enum class (same micronaut package) with custom statuses added on your project source directory. If you distribute your application as a "fat jar", your local class will overwrite framework's class.

It's not as elegant as it should be, but it works and you can start using your custom statuses.

Either way, I think the possibility of using custom statuses would be a nice add to the framework.